bot/exts/info/doc/_cog.py

```diff
@@ -0,0 +1,467 @@
+    async def docs_group(self, ctx: commands.Context, *, symbol: Optional[str]) -> None:
+        """Lookup documentation for Python symbols."""
+        await ctx.invoke(self.get_command, symbol=symbol)
+
+    @docs_group.command(name='getdoc', aliases=('g',))
+    async def get_command(self, ctx: commands.Context, *, symbol: Optional[str]) -> None:
+        """
+        Return a documentation embed for a given symbol.
+
+        If no symbol is given, return a list of all available inventories.
+
+        Examples:
+            !docs
+            !docs aiohttp
+            !docs aiohttp.ClientSession
+            !docs getdoc aiohttp.ClientSession
+        """
+        if not symbol:
+            inventory_embed = discord.Embed(
+                title=f"All inventories (`{len(self.base_urls)}` total)",
+                colour=discord.Colour.blue()
+            )
+
+            lines = sorted(f"• [`{name}`]({url})" for name, url in self.base_urls.items())
+            if self.base_urls:
+                await LinePaginator.paginate(lines, ctx, inventory_embed, max_size=400, empty=False)
+
+            else:
+                inventory_embed.description = "Hmmm, seems like there's nothing here yet."
+                await ctx.send(embed=inventory_embed)
+
+        else:
+            symbol = symbol.strip("`")
+            # Fetching documentation for a symbol (at least for the first time, since
+            # caching is used) takes quite some time, so let's send typing to indicate
+            # that we got the command, but are still working on it.
+            async with ctx.typing():
+                doc_embed = await self.get_symbol_embed(symbol)
+
+            if doc_embed is None:
+                symbol = await discord.ext.commands.clean_content().convert(ctx, symbol)
+                error_embed = discord.Embed(
+                    description=f"Sorry, I could not find any documentation for `{(symbol)}`.",
+                    colour=discord.Colour.red()
+                )
+                error_message = await ctx.send(embed=error_embed)
+                await wait_for_deletion(
+                    error_message,
+                    (ctx.author.id,),
+                    timeout=NOT_FOUND_DELETE_DELAY,
+                    client=self.bot
+                )
+                with suppress(discord.NotFound):
+                    await ctx.message.delete()
+                with suppress(discord.NotFound):
+                    await error_message.delete()
+            else:
+                msg = await ctx.send(embed=doc_embed)
+                await wait_for_deletion(msg, (ctx.author.id,), client=self.bot)
+
+    @docs_group.command(name='setdoc', aliases=('s',))
+    @commands.has_any_role(*MODERATION_ROLES)
+    async def set_command(
+        self, ctx: commands.Context, package_name: PackageName,
+        base_url: ValidURL, inventory_url: InventoryURL
```

Comment:
I'm not entirely sold on this being worth it with the rare usage of the command, using the inventory dict here would mean having to create one more method to nicely handle the argument before passing it over to `update_single`

Reply:
It doesn't seem that complicated to me. Return a tuple from the converter and change `update_single` to accept an inventory instead of a URL. Change any other callers to fetch the inventory instead of passing a URL.

Reply:
Implemented it in https://github.com/python-discord/bot/pull/1014/commits/5258c797e1ea3bc08eeddb5fd30dc35b85628e3e after consideration separating the rescheduling logic from the update method is also nice. As a consequence the fetches from the set command now don't get rescheduled but I don't think that's much of an issue